exec/java-exec/src/main/java/org/apache/drill/exec/planner/fragment/Stats.java

```diff
@@ -43,6 +43,12 @@ public void addMinWidth(int minWidth) {
     collector.addMinWidth(minWidth);
   }
 
+  public void setDistributionAffinity(final DistributionAffinity distributionAffinity) {
+    if (this.distributionAffinity.isLessOrEqualRestrictive(distributionAffinity)) {
```

Comment:
I guess if the affinity was lesser (not equals) only, it would work too (equals means that's the same policy, so the assignment is a no-op)